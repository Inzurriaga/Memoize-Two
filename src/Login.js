import React, { Component } from "react"
import SignIn from "./SignIn"
import CreateAccount from "./CreateAccount"

export default class login extends Component{
    constructor(){
        super();
        this.state = {
            toggleLogin: "rotateY(0deg)"
        }
    }

    toggleLoginOption = () => {
        if( this.state.toggleLogin === "rotateY(0deg)"){
            this.setState({
                toggleLogin: "rotateY(180deg)"
            })
        }else{
            this.setState({
                toggleLogin: "rotateY(0deg)"
            })
        }
    }

    render() {
        return(
            <div className="login-section" style={{transform: this.state.toggleLogin }}>
               <SignIn toggle={this.toggleLoginOption} 
                        users={this.props.users} 
                        loginToAccount={this.props.loginToAccount}/>
               <CreateAccount toggle={this.toggleLoginOption}
                            createNewAccount={this.props.createNewAccount} users={this.props.users}/>
            </div>
        )
    }
}