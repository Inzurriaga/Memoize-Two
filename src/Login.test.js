import React from 'react';
import { shallow } from 'enzyme';
import Login from "./Login"

const mockCreateNewAccout = jest.fn()
const mockUsers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[1,2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}]
const mockLoginToAccount = jest.fn()

describe("Login", ()=> {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Login createNewAccount={mockCreateNewAccout} 
              users={mockUsers} 
              loginToAccount={mockLoginToAccount}/>
    )
  })
  it("should  match Snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should have a default state", () => {
    expect(wrapper.state()).toEqual({
        toggleLogin: "rotateY(0deg)"
    })
  })

  it("should have state change when toggle function is invoke", () => {
    expect(wrapper.state()).toEqual({
      toggleLogin: "rotateY(0deg)"
    })
    wrapper.instance().toggleLoginOption()
    expect(wrapper.state()).toEqual({
      toggleLogin: "rotateY(180deg)"
    })
  })
})