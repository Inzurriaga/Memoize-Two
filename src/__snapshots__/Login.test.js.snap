// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should  match Snap shot 1`] = `
<div
  className="login-position"
>
  <div
    className="login-section"
    style={
      Object {
        "transform": "rotateY(0deg)",
      }
    }
  >
    <SignIn
      loginToAccount={[MockFunction]}
      toggle={[Function]}
      users={
        Array [
          Object {
            "completed": Array [
              1,
              2,
              5,
            ],
            "name": "duy",
            "password": "duy",
          },
          Object {
            "completed": Array [
              1,
              2,
              5,
            ],
            "name": "gabriel",
            "password": "gabriel",
          },
          Object {
            "completed": Array [
              1,
              2,
            ],
            "name": "isaac",
            "password": "isaac",
          },
        ]
      }
    />
    <CreateAccount
      createNewAccount={[MockFunction]}
      toggle={[Function]}
      users={
        Array [
          Object {
            "completed": Array [
              1,
              2,
              5,
            ],
            "name": "duy",
            "password": "duy",
          },
          Object {
            "completed": Array [
              1,
              2,
              5,
            ],
            "name": "gabriel",
            "password": "gabriel",
          },
          Object {
            "completed": Array [
              1,
              2,
            ],
            "name": "isaac",
            "password": "isaac",
          },
        ]
      }
    />
  </div>
</div>
`;
